pkg/hpa/service.go

```diff
@@ -765,3 +770,44 @@ func (c *Service) excludeExternalMetric(ctx context.Context, hpa *v2.HorizontalP
 
 	return newHPA
 }
+
+func (c *Service) CheckHpaMetricStatus(ctx context.Context, currenthpa *v2.HorizontalPodAutoscaler) bool {
+	//currenthpa = currenthpa.DeepCopy()
+	logger := log.FromContext(ctx)
+	if currenthpa == nil {
+		logger.Info("empty HPA passed into status check, ignore")
+		return true
+	}
+
+	if reflect.DeepEqual(currenthpa.Status, v2.HorizontalPodAutoscalerStatus{}) || currenthpa.Status.Conditions == nil || currenthpa.Status.CurrentMetrics == nil {
```

Comment:
```suggestion
	if currenthpa==nil || reflect.DeepEqual(currenthpa.Status, v2.HorizontalPodAutoscalerStatus{}) || len(currenthpa.Status.Conditions) != 0 || len(currenthpa.Status.CurrentMetrics) != 0 {
		// shouldn't reach here because, in this HPA unready case, the controller should stop the reconciliation at the point of fetching hpa.
		logger.Error("unready HPA is passed to IsHpaMetricAvailable")
		return false
```